core/Tracker/Response.php

```diff
@@ -161,6 +172,40 @@ private function outputTransparentGif()
         echo base64_decode($transGifBase64);
     }
 
+    /**
+     * Output a custom tracking image
+     *
+     * @param string $customImage The custom image setting specified in the config
+     * @return bool True if the custom image was successfully output, else false
+     */
+    private function outputCustomImage(string $customImage): bool
+    {
+
+        if (strlen($customImage) > 2 && substr($customImage, -2) == '==') {
+
+            $transGifBase64 = $customImage;
+            Common::sendHeader('Content-Type: image/gif');
+
+            echo base64_decode($transGifBase64);
+            return true;
+        }
+
+        if (file_exists($customImage)) {
```

Comment:
not too important but might want to use `is_file` as it's bit faster see https://github.com/matomo-org/matomo/issues/16552 . It usually wouldn't make a big difference unless someone tracks heaps of traffic. Technically would also need to check `is_readable`.